src/main/java/com/example/controller/settings/LamportSettingsController.java

```diff
@@ -50,6 +50,8 @@ public void adjustSettingsConditions(MyGraph<Integer, Integer> graph) {
         Vertex<Integer> commander = graph.vertices().stream().toList().get(0);
         int maxDepth = graph.getLongestPathFor(commander);
         algorithmSettings.getSettings().get("depth").setValidateArgument((value) -> (Integer) value >= 0 && (Integer) value <= maxDepth);
+        AlgorithmSetting<Integer> depthSetting = (AlgorithmSetting<Integer>) algorithmSettings.getSettings().get("depth");
+        depthSetting.setValue(maxDepth);
```

Comment:
I am not sure, but proposed depth should be equal to the number of traitors. Please ask Agnieszka to clarify it.

Reply:
It should be min value of traitors and max depth.